### ISSUE TITLE
Batch DELETE throws batch-response-parser error

### ISSUE THREAD

**Issue by stefania-santimbrean** [SAP/cloud-sdk-js#939]
Hello,

I am doing a batch request which contains DELETE requests for PurchaseOrderItems using the following code:

```
public async deletePurchaseOrderItems(purchaseOrderId: string, purchaseOrderItemIds: string[]): Promise<void> {
    try {
        const deleteRequests = purchaseOrderItemIds.map(poi =>
            PurchaseOrderItem.requestBuilder().delete(purchaseOrderId, poi)
        )
        // Execute the batch request
        await batch(changeset(...deleteRequests)).execute(Constants.DESTINATION);
        this.logger.debug(`Purchase Order ${purchaseOrderId} items deleted`);
    } catch (err) {
        this.logger.debug(err);
    }
}
```
The request does what I need, which is to mark for deletion the items in the Purchase Order, but the logs show an error:
![image](https://user-images.githubusercontent.com/74544010/105845156-ff92f200-5fe2-11eb-9a65-23abcbbed18a.png)

Could you please help me figure out where is the error coming from an how to handle it?

**Comment by jjtang1985** [comment]
Hi @stefania-santimbrean , thank you for approaching us.

From the log statement, I assume this is either because of the unexpected response format from the service or there is a bug from the sdk.
Is it possible to provide the response so that we can understand better?

Please be aware, I see logs with both `error` and `warn` levels, which might come from different cases.
Therefore, if you see batch response with different format, please share them together.

Thank you very much.
Junjie

**Comment by stefania-santimbrean** [comment]
Hello @jjtang1985 ,

Thanks for responding so quick!

I changed the code to:
```
// Execute the batch request
const response = await batch(changeset(...deleteRequests)).execute(Constants.DESTINATION);
this.logger.debug(response);
```
The logged response object looks like this:
```
[
  {
    "responses": [
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      },
      {
        "body": {},
        "httpCode": 204
      }
    ]
  }
]
```
Snapshot of response object from debugger maybe is better:
![image](https://user-images.githubusercontent.com/74544010/105892410-7f887e80-601a-11eb-9e8c-63528c2f9b85.png)

Please let me know what else can I do to further provide information about the issue.

Thank you,
Kind regards,
Stefania

**Comment by jjtang1985** [comment]
Hi @stefania-santimbrean , thank you for your example.

Theoretically, `204` means the response body should be empty, but the error was `Unexpected token d...`.
I guess the raw response body contains some lines or words starting with a `d`.

May I ask you to debug further via e.g., postman so that we can see the complete raw response from the service?

By the way, I assume you are using this pre-built client `@sap/cloud-sdk-vdm-purchase-order`.

**Comment by stefania-santimbrean** [comment]
Hello @jjtang1985 ,

I tested now the BATCH delete with the same payload in Postman and this is the response:
```
--3B17E95920A7FAF8BCB7495D043515000
Content-Type: multipart/mixed; boundary=3B17E95920A7FAF8BCB7495D043515001
Content-Length:      2427

--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001
Content-Type: application/http
Content-Length: 71
content-transfer-encoding: binary

HTTP/1.1 204 No Content
Content-Length: 0
dataserviceversion: 2.0


--3B17E95920A7FAF8BCB7495D043515001--

--3B17E95920A7FAF8BCB7495D043515000--
```

I attach also the versions I use:
```
"@sap-cloud-sdk/core": "^1.35.0",
"@sap/cloud-sdk-vdm-purchase-order-service": "^1.23.0",
```

Please let me know what else can I do.

Thank you so much,
Stefania

**Comment by jjtang1985** [comment]
Hi @stefania-santimbrean,
Thank you for your raw response example.

I can reproduce your error by using the example provided.
This is a bug from the SDK and we will work on it.

Please correct me.
From my understanding, the batch request is executed without any issues, meaning the delete operations are successfully applied.
The response object looks also fine, since `204` should not return any real data as body.
The issue is about the false-positive error message.

Best regards,
Junjie

**Comment by stefania-santimbrean** [comment]
Hello @jjtang1985 ,

Yes, you are right, the issue is about the error & warn logs. I rely on the logging for the app (someone might look at the logs often to make sure the app does its job).

Do you happen to know when you can release a fix?

Thank you so much for the support,
Best regards,
Stefania

**Comment by jjtang1985** [comment]
Hi @stefania-santimbrean , 
the backlog item was refined and we might start working on it next week, but there is no guarantee.

Could you please share whether the issue is related a productive application, staging environment or maybe PoC project?

Thank you.

Best regards,
Junjie

**Comment by stefania-santimbrean** [comment]
Hello @jjtang1985 .

It's intended for a productive application that I developed and is going into testing now.

Best regards,
Stefania

**Comment by jjtang1985** [comment]
Hi @stefania-santimbrean ,

I'll share this information with the team and will let you know once there are some updates.

Best regards,
Junjie

**Comment by stefania-santimbrean** [comment]
Hello @jjtang1985 ,

Thanks so much for all the support!!

Best regards,
Stefania

**Comment by tomfrenken** [comment]
Hey @stefania-santimbrean, we have solved the issue and it should be available to you in the next release, please let us now if the issue was solved for you.

**Comment by jjtang1985** [comment]
Hi @stefania-santimbrean , 
we released a new version [1.40.0](https://github.com/SAP/cloud-sdk-js/releases) this morning.
I'll close the ticket.
Please feel free to reopen or comment in case you have further questions.

By the way, we release a `canary` version to npm for every commit, which means actually you could already test the `canary` version when the [PR](https://github.com/SAP/cloud-sdk-js/pull/1074) was merged.

**Comment by stefania-santimbrean** [comment]
@tomfrenken @jjtang1985 ,

Thanks so much for the support with the issue! Just got around to test it and it works, sorry for the delay :)
